### PR TITLE
Remove check on dimensioning symbols (towards #1089)

### LINKED ISSUE #1089: [psyir] Check on whether dimensioning symbol is local does not work when symbol is in an ancestor scope.
For the following Fortran:

    module my_mod
      integer :: dim
    contains
      subroutine init()
        dim = 10
      end subroutine
      subroutine work()
        real, dimension(dim) :: my_array
      end subroutine
    end module

Then PSyclone complains:

    If a local datasymbol is used as part of a dimension declaration then it should be a constant, but 'dim' is not.

The problem is in `ArrayType._validate_shape` where the Symbol being referred to in the shape specification is looked up. It is found to have a 'local' interface and not be a constant. Both of which are correct. However, no check is made on which SymbolTable the symbol is in.

### PR DESCRIPTION
**Pull request by arporter**
This PR simply removes the current, incorrect, check. However, with some work on the naming/definition of symbol interfaces it should be possible to put back a correct version - hence this PR does not close #1089.

### CODE REVIEW

**Comment by arporter** [comment]
This PR simply removes the check that dimensioning symbols are not local and not constant. (Since we currently actually have no way of reliably checking whether or not a symbol is local.)

**Comment by arporter** [comment]
Once this PR is on, PSyclone can process every file bar one in the NEMO oce+ice configuration we're using. The one that we can't process (`obs_utils.f90`) is actually because fparser can't parse it - there's an access of a 1D array named `idim` that fparser incorrectly identifies as a call to the IDIM intrinsic which should have two arguments. Since it's only a 1D array, fparser reports a syntax error.

**Review by sergisiso — CHANGES_REQUESTED** [review]
@arporter I am not totally convinced that the checks couldn't be done if the interfaces were improved as discussed in the issue comments but I appreciate these need to be removed for now.

I propose renaming this PR to "towards 1089" instead of "closes", adding a TODO where the checking code was, and bring the removed tests back but mark them both with an xfail to this issue number. Would that be OK?

File: src/psyclone/psyir/symbols/datatypes.py [hunk 1]
```diff
@@ -495,7 +482,7 @@ def add(self, name, datatype, visibility):
         '''
         # This import has to be here to avoid circular dependencies
         # pylint: disable=import-outside-toplevel
-        from psyclone.psyir.symbols import Symbol, TypeSymbol
+        from psyclone.psyir.symbols import Symbol
```
[the remark this answers is not in the document]
Comment (arporter) [reply src/psyclone/psyir/symbols/datatypes.py:485]: Oh yes, if I import it from `psyir.symbols.symbol` then we avoid the circular import.

**Comment by arporter** [comment]
I've done as you suggested and put back the test I'd removed, along with a comment in the source file itself. Note that one of the tests was outdated so it has still been deleted. Hopefully ready for another look.

**Review by sergisiso — APPROVED** [review]
@arporter Thanks for taking the suggestion, I think the distinction of a local and a global used as a local will need to be done at some point, and then we can revisit this check.

The PR is ready. It mostly remove code lines, all lines are covered and tests pass. There are no formatting/styling issues. Documentation is not necessary. There are new TODOs to the issue as the issue is not going to be closed.

**Pull request merged**